client-spark/common/src/main/java/org/apache/spark/shuffle/writer/AddBlockEvent.java

```diff
@@ -61,16 +54,30 @@ public List<Runnable> getProcessedCallbackChain() {
     return processedCallbackChain;
   }
 
-  public void withBlockSentCallback(TupleConsumer<ShuffleBlockInfo, Boolean> blockSentCallback) {
-    this.blockSentCallback = blockSentCallback;
+  @Override
+  public String toString() {
+    return "AddBlockEvent: TaskId[" + taskId + "], " + shuffleDataInfoList;
   }
 
-  public TupleConsumer<ShuffleBlockInfo, Boolean> getBlockSentCallback() {
-    return blockSentCallback;
+  public void callbackOnBlockFailure(ShuffleBlockInfo block) {
+    if (block == null) {
+      return;
+    }
+    this.blockFailureCallback.onBlockFailure(block);
   }
 
-  @Override
-  public String toString() {
-    return "AddBlockEvent: TaskId[" + taskId + "], " + shuffleDataInfoList;
+  public void callbackOnBlockSuccess(ShuffleBlockInfo block) {
```

Comment:
The name of the method seems weird. Let us think a better name.

Reply:
done